src/domain/useCase/deleteService.go

```diff
@@ -4,16 +4,82 @@ import (
 	"errors"
 	"log"
 
+	"github.com/speedianet/os/src/domain/entity"
 	"github.com/speedianet/os/src/domain/repository"
 	"github.com/speedianet/os/src/domain/valueObject"
 )
 
-func DeleteService(
+type DeleteService struct {
+	queryRepo      repository.ServicesQueryRepo
+	cmdRepo        repository.ServicesCmdRepo
+	vhostQueryRepo repository.VirtualHostQueryRepo
+	vhostCmdRepo   repository.VirtualHostCmdRepo
+}
+
+func NewDeleteService(
 	queryRepo repository.ServicesQueryRepo,
 	cmdRepo repository.ServicesCmdRepo,
+	vhostQueryRepo repository.VirtualHostQueryRepo,
+	vhostCmdRepo repository.VirtualHostCmdRepo,
+) DeleteService {
+	return DeleteService{
+		queryRepo:      queryRepo,
+		cmdRepo:        cmdRepo,
+		vhostQueryRepo: vhostQueryRepo,
+		vhostCmdRepo:   vhostCmdRepo,
+	}
+}
+
+func (uc DeleteService) deleteSvcAutoMapping(
 	svcName valueObject.ServiceName,
 ) error {
-	serviceEntity, err := queryRepo.GetByName(svcName)
+	vhostsWithMappings, err := uc.vhostQueryRepo.GetWithMappings()
+	if err != nil {
+		return err
+	}
+
+	if len(vhostsWithMappings) == 0 {
+		return nil
+	}
+
+	primaryVhostWithMapping := vhostsWithMappings[0]
```

Comment:
Você assume que o primeiro item do vhostsWithMapping é o primary. Você consegue ter certeza disso? Amanhã vamos mudar pra SQLite. E aí? Não há como confiar na posição do elemento cara.

Reply:
Ajustado.
Agora é feito um for lá dentro em busca do primaryVhost.